### PR TITLE
Introducing mesh layers

### PR DESCRIPTION
**Pull request by DelazJ**
### Description

Goal: This backports the description of the mesh layer, with some review

Ticket: There might/should be but i can't really find which one.

- [ ] Backport to LTR documentation is required (No backport needed but third commit and possible review commit(s) will need to be forward ported)

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

**Comment by rduivenvoorde** [comment]
I'd wait for @rosaguilar to finish the chapter I think before backporting (or copying?) it to 3.4

But feel free to merge now.

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+ Working with Mesh Data
+***********************
+
+.. toctree::
+   :maxdepth: 2
+
+Mesh properties
+===============
+
+A mesh is an unstructured grid usually with temporal and other component.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
A mesh is an unstructured grid usually with temporal and other components.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+The spatial component contains a collection of vertices, edges and faces
+in 2D or 3D space:
+
+* **vertices** - XY(Z) points (in the layer's coordinate reference system)
+* **edges** - connect pairs of vertices
+* **faces** - sets of edges forming a closed shape - typically triangles or
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
* **faces** - a face is a set of edges forming a closed shape - typically a triangle or
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  quadrilaterals (quads), rarely polygons with higher number of vertices
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
  a quadrilateral (quad), rarely polygons with more vertices
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+
+.. _figure_mesh_grid_types:
+
+.. figure:: img/mesh_grid_types.png
+   :align: center
+
+   Different mesh types
+
+QGIS can render mesh data using a triangle grid or a regular squad mesh.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
QGIS can render mesh data using triangles or quadrants.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+For example, having a triangular mesh with numbered vertices as shown in the image below:
+
+.. _figure_triangual_grid_with_numered_vertices:
+
+.. figure:: img/triangual_grid_with_numered_vertices.png
+   :align: center
+
+   Triangular grid with numbered vertices
+
+Each vertex can store different datasets -typically multiple quantities-,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
Each vertex can store different datasets (typically multiple quantities),
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+and those datasets can also have a temporal dimension. Thus, a single file may
+contain multiple datasets.
+
+The following table gives an idea what information is stored in datasets.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
The following table gives an idea about the information that can be stored in mesh datasets.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+Table columns represent indices of mesh vertices, each row represents one dataset.
+Datasets can have different datatypes.
+In this case, it stored the wind velocity at 10m at particular moment of time (t1, t2, t3).
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
In this case, it stores wind velocity at 10m at a particular moments in time (t1, t2, t3).
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+In a similar way, the mesh dataset can store vector values at each vertex,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
In a similar way, the mesh dataset can store vector values for each vertex,
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+for example, vector of wind direction at given time stamps:
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
for example, a wind direction vector at the given time stamps:
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+=============================== ========= ========= ========= =====
+
+.
+
+We can visualize the data by assigning colors to values (similarly to how it is
+done with :ref:`Singleband pseudocolor <label_colormaptab>` raster rendering)
+and interpolating data between vertices according to the mesh topology.
+It is common that some quantities are 2D vectors rather than being
+simple scalar values (e.g. wind direction).
+For such quantities it is very desired to display arrows indicating direction.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
For such quantities it is desirable to display arrows indicating direction.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+
+QGIS accesses mesh data using the `MDAL drivers <https://github.com/lutraconsulting/MDAL>`_.
+Hence, the natively supported formats are:
+
+* ``NetCDF``: Generic format for scientific data
+* ``GRIB``: Format commonly used in meteorology
+* ``XMDF``: As an example, hydraulic outputs from TUFLOW modelling package
+* ``DAT``: Outputs of various hydrodynamic modelling packages (e.g. BASEMENT, HYDRO_AS-2D, TUFLOW)
+* ``3Di``: 3Di modelling package format based on Climate and Forecast Conventions (http://cfconventions.org/)
+* Some examples of mesh datasets can be found: https://apps.ecmwf.int/datasets/data/interim-full-daily/levtype=sfc/
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
* Some examples of mesh datasets can be found at: https://apps.ecmwf.int/datasets/data/interim-full-daily/levtype=sfc/
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+
+Loading a mesh layer
+====================
+
+To add a mesh layer to QGIS, open the :menuselection:`Data Source Manager`
+dialog, either by selecting it from the :menuselection:`Layer -->` menu or
+clicking the |dataSourceManager| :sup:`Open Data Source Manager` button.
+Then, enable the |addMeshLayer| :guilabel:`Mesh` tab on the left panel
+and press the :guilabel:`...` button to select the file and press
+:guilabel:`Add`. The layer will be added using the native mesh rendering.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
:guilabel:`Add`.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+.. _figure_mesh_info_properties:
+
+.. figure:: img/mesh_info_properties.png
+   :align: center
+
+   Mesh Layer Properties
+
+The :guilabel:`Information` tab is read-only and represents an interesting
+place to quickly grab summarized information and metadata on the current layer.
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
place to quickly grab summarized information and metadata for the current layer.
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+
+Source
+------
+
+The :guilabel:`Source` tab displays basic information about the selected mesh, including:
+
+* the Layer name to display in the :guilabel:`Layers` panel
+* setting the Coordinate Reference System: Displays the layer’s Coordinate
+  Reference System (CRS) as a PROJ.4 string. You can change the layer’s CRS,
+  selecting a recently used one in the drop-down list or clicking on |setProjection|
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
  by selecting a recently used one in the drop-down list or clicking on |setProjection|
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+face count and dataset groups count.
+
+Source
+------
+
+The :guilabel:`Source` tab displays basic information about the selected mesh, including:
+
+* the Layer name to display in the :guilabel:`Layers` panel
+* setting the Coordinate Reference System: Displays the layer’s Coordinate
+  Reference System (CRS) as a PROJ.4 string. You can change the layer’s CRS,
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
  Reference System (CRS) as a PROJ.4 string. You can change the layer’s CRS
```

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+  selecting a recently used one in the drop-down list or clicking on |setProjection|
+  :guilabel:`Select CRS` button (see :ref:`crs_selector`).
+  Use this process only if the CRS applied to the layer is a wrong one or
```
Comment (havatv) [inline source/docs/user_manual/working_with_mesh/mesh_properties.rst]:
```suggestion
  Use this process only if the CRS applied to the layer is wrong or
```

Comment (rosaguilar) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: Thanks!

**Comment by rduivenvoorde** [comment]
@havatv thanks for suggestions. Can @rosaguilar handle this? @rosaguilar do you also see the  Commit suggestion button? Or should I apply the suggestions?

gnome-shell-screenshot-14LBYZ

**Comment by DelazJ** [comment]
@rduivenvoorde normally she shouldn't. Only available to PR owner and repo committers I think. I can't review this now so feel free to do.
Btw, better use the "Add suggestion to batch" button in the files tab to commit all accepted suggestion in a single commit
image

**Comment by PeterPetrik** [comment]
thanks for doing this!

**Comment by rosaguilar** [comment]
> @rduivenvoorde normally she shouldn't. Only available to PR owner and repo committers I think. I can't review this now so feel free to do.
> Btw, better use the "Add suggestion to batch" button in the files tab to commit all accepted suggestion in a single commit
> image

Indeed. I can't apply the suggestions. I don't see the button.

**Comment by rduivenvoorde** [comment]
@rosaguilar then I'll just apply all suggestions and pull

**Comment by rosaguilar** [comment]
> I'd wait for @rosaguilar to finish the chapter I think before backporting (or copying?) it to 3.4
> 
> But feel free to merge now.

I have finished 

> @rosaguilar then I'll just apply all suggestions and pull

@rduivenvoorde do maar. I have almost finished the rest of the text. Maybe in half an hour is ready.

**Comment by rduivenvoorde** [comment]
@DelazJ @rosaguilar I cannot push this (I tried in the files tab), apparently the PR-holder should do: @DelazJ

**Comment by DelazJ** [comment]
> @DelazJ @rosaguilar I cannot push (neither accept the suggestions) this (I tried in the files tab), apparently the PR-holder should do: @DelazJ

Sorry I thought you could. Weird; I allowed maintainers modification though.

**Pull request merged**